projects/components/src/modal/modal.ts

```diff
@@ -19,6 +19,33 @@ export const enum ModalSize {
   MediumWide = 'medium-wide'
 }
 
+export interface ModalDimension {
+  // Number => without unit (considered px) and String => with units (expression included)
+  width: number | string;
```

Comment:
What if I pass `23sdgsg` ?

Reply:
I mean, it won't work at that time. And I think this should be the case. Like suppose if we do this in the css (putting wrong value for width), then it doesn't work. So It is developer's responsibility to put correct value and I think developer can handle this.

Reply:
If you want to get really fancy, TS supports this kind of thing now. I'd also be fine just taking in any string, we don't validate to nearly this level elseewhere.

```ts
type Unit = "px" | "em" | "fr" | "%"; // probably some others too

type SizeWithUnit = `${number}${Unit}`;
```

Reply:
in action: 
![image](https://user-images.githubusercontent.com/45047841/131523838-c6fff579-2f50-45ee-9c68-00ec12bc1b3a.png)

Reply:
Yeah, we could do this, but my intention of using string, to use these kind of expression as well, not only simple units - `calc(100vh - 100px)`

Reply:
```ts
type SizeExpression = SizeWithUnit | `calc(${SizeWithUnit} - ${SizeWithUnit})`
```

Typescript type system is turing complete 😉 

But seriously, yes I agree, overkill.

Reply:
Yeah!!!

Reply:
I will let you guys decide :D

Reply:
According to me, It is good enough and as @aaron-steinfeld  said doing this, is an overkill.
I think we can merge this.